apps/server/server.cpp

```diff
@@ -314,13 +320,16 @@ void invoke_sdk_api(payload::ClientRequest buff_recv) {
             buff_send.set_status(::payload::Status::UNREACHABLE);
             break;
         }
-        camDepthSensor = depthSensors.front();
-        aditof::SensorDetails depthSensorDetails;
-        camDepthSensor->getDetails(depthSensorDetails);
+
+        int depth_sensor_id = 0;
         auto pbSensorsInfo = buff_send.mutable_sensors_info();
-        sensorV4lBufAccess =
-            std::dynamic_pointer_cast<aditof::V4lBufferAccessInterface>(
-                camDepthSensor);
+        for (const auto &depthSensor : depthSensors) {
+            camDepthSensor[depth_sensor_id] = depthSensor;
```

Comment:
depth_sensor_id should be incremented in this loop. Otherwise it will always be 0.

Reply:
yes, thank you! fixed